provider/oci/networking_test.go

```diff
@@ -0,0 +1,246 @@
+		map[string]string{
+			"id":        vnicID,
+			"life":      string(ociCore.VnicLifecycleStateAvailable),
+			"name":      "fakeVnicName",
+			"privateIP": "1.1.1.1",
+			"publicIP":  "2.2.2.2",
+		},
+	})
+
+	vcnRequest, vcnResponse := makeListVcnRequestResponse(
+		n.testCompartment, n.tags, []map[string]string{
+			map[string]string{
+				"cidr":    "1.0.0.0/8",
+				"id":      vcnID,
+				"life":    string(ociCore.VcnLifecycleStateAvailable),
+				"route":   "fakeRouteTable",
+				"seclist": "fakeSeclist",
+				"name":    "amazingVcn",
+			},
+		})
+
+	subnetRequest, subnetResponse := makeListsubnetsRequestResponse(
+		n.testCompartment, "fakeVcn", []map[string]string{
+			map[string]string{
+				"name":  "fakeSubnet",
+				"az":    "us-phoenix-1",
+				"cidr":  "1.0.0.0/8",
+				"id":    "fakeSubnetId",
+				"vcnid": vcnID,
+				"route": "fakeRouteTable",
+				"life":  string(ociCore.SubnetLifecycleStateAvailable),
+			},
+			map[string]string{
+				"name":  "fakeSubnet",
+				"az":    "us-phoenix-1",
+				"cidr":  "1.0.0.0/8",
+				"id":    "anotherFakeSubnetId",
+				"vcnid": vcnID,
+				"route": "fakeRouteTable",
+				"life":  string(ociCore.SubnetLifecycleStateAvailable),
+			},
+		})
+
+	request, response := makeGetInstanceRequestResponse(
+		n.testCompartment, n.tags, map[string]string{
+			"id":    n.testInstanceID,
+			"name":  "fake",
+			"az":    "us-phoenix-1",
+			"shape": "VM.Standard1.1",
+		}, ociCore.InstanceLifecycleStateRunning)
+
+	gomock.InOrder(
+		n.netw.EXPECT().ListVcns(context.Background(), vcnRequest).Return(vcnResponse, nil),
+		n.netw.EXPECT().ListSubnets(context.Background(), subnetRequest).Return(subnetResponse, nil),
+		n.compute.EXPECT().GetInstance(context.Background(), request).Return(response, nil).Times(1),
+		n.compute.EXPECT().ListVnicAttachments(context.Background(), attachRequest).Return(attachResponse, nil),
+		n.netw.EXPECT().GetVnic(context.Background(), vnicRequest[0]).Return(vnicResponse[0], nil),
+
+		n.netw.EXPECT().ListVcns(context.Background(), vcnRequest).Return(vcnResponse, nil),
+		n.netw.EXPECT().ListSubnets(context.Background(), subnetRequest).Return(subnetResponse, nil),
+		n.compute.EXPECT().GetInstance(context.Background(), request).Return(response, nil).Times(1),
+		n.compute.EXPECT().ListVnicAttachments(context.Background(), attachRequest).Return(attachResponse, nil),
+		n.netw.EXPECT().GetVnic(context.Background(), vnicRequest[0]).Return(vnicResponse[0], nil),
+	)
+	lookFor := []network.Id{
+		network.Id("fakeSubnetId"),
+	}
+	info, err := n.env.Subnets(nil, instance.Id(n.testInstanceID), lookFor)
+	c.Assert(err, gc.IsNil)
+	c.Check(info, gc.HasLen, 1)
+	c.Assert(info[0].CIDR, gc.Equals, "1.0.0.0/8")
+
+	lookFor = []network.Id{
+		network.Id("notHere"),
+	}
+	_, err = n.env.Subnets(nil, instance.Id(n.testInstanceID), lookFor)
+	c.Check(err, gc.ErrorMatches, "failed to find the following subnet ids:.*notHere.*")
+}
```

Comment:
> I do think we're missing some test coverage here. Things like how we allocate subnets don't seem to be tested, and how we initialize a VCN.

this will be tested indirectly in the next PR (6/7). Without creating package tests, there is no way to test private methods.

Will look at switching the ControllerUUID with model UUIDs. However, this means we will be creating 3 subnets per model. Currently the cleanup code for the networking bits only runs when the controller gets killed, hence the reason for using only the controller label.

Creating 3 subnets per model, in a scenario that might have dozens of models created, will probably be a bit messy. I have to check if OCI has quotas regarding subnets. as well.

Reply:
SGTM